internal/server/server.go

```diff
@@ -162,17 +164,10 @@ func InitializeConfigs(ctx context.Context, cfg ServerConfig, l log.Logger, inst
 }
 
 // NewServer returns a Server object based on provided Config.
-func NewServer(ctx context.Context, cfg ServerConfig, l log.Logger) (*Server, error) {
-	instrumentation, err := CreateTelemetryInstrumentation(cfg.Version)
-	if err != nil {
-		return nil, fmt.Errorf("unable to create telemetry instrumentation: %w", err)
-	}
-
+func NewServer(ctx context.Context, cfg ServerConfig, l log.Logger, instrumentation *Instrumentation) (*Server, error) {
```

Comment:
Why does instrumentation need to be passed in here?

Should  logger / instrumentation be moved into the context?